src/main/java/net/dries007/tfc/world/chunkdata/ChunkData.java

```diff
@@ -222,6 +224,23 @@ public void generateFull(int[] surfaceHeight, int[] aquiferSurfaceHeight)
         this.status = Status.FULL;
     }
 
+    public void modifyBaseGroundwater(int[] surfaceHeight, ChunkAccess chunk)
+    {
+        assert this.baseGroundwaterLayer != null;
+        float groundwater00 = modifyBaseGroundwaterPoint(surfaceHeight[0], this.baseGroundwaterLayer.value00());
```

Comment:
would rather these constants get written out as the source x/y pair, i.e. `0 + 16 * 0`, `0 + 16 * 15`, `15 + 16 * 16`, etc. Alternatively use an `index(int x, int z) { return x + 16 * z; }` function if you don't want the IDE to complain about computed constants.

Reply:
I've just added comments because I just feel bad making the game do more math than it needs to